File: Data Structures & Algorithms/coin-change/submission-0.java
class Solution {
    public int coinChange(int[] coins, int amount) 
    {
        
        int ans = coinChangeHelper(coins,amount,coins.length-1);
        return ans == Integer.MAX_VALUE?-1:ans;
        
    }
    public int coinChangeHelper(int[] coins, int target, int index){
        if(target == 0) return 0;
        if(index < 0 || target<0) return Integer.MAX_VALUE;
        int added = 0;
        int not_added = coinChangeHelper(coins,target,index-1);
        
        added = coinChangeHelper(coins,target-coins[index],index);
        if(added != Integer.MAX_VALUE){
                added++;
        }
        return Math.min(added,not_added);
         
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        ans = [0]*(n)
        ans[0] = cost[0]
        ans[1] = cost[1]
        for i in range(2,n):
            ans[i] = cost[i] + min(ans[i-1],ans[i-2])
        return min(ans[n-1],ans[n-2])
        